proposed/2022/PMUIPackageSourceMapping.md

```diff
@@ -0,0 +1,72 @@
+# PMUI 
+* Start Date: 6/25/22
+* Authors: Ella McNally
+* Issue
+
+## Summary
+
+Currently there is no support for package source mapping in the PMUI. I will add a feature that allows the user to choose a source to map a package to when they install it and a feature that shows all package source mappings in the preview. This feature is a portion of what was discussed in a previous spec. https://github.com/NuGet/Home/blob/dev/implemented/2021/PackageSourceMapping.md
+
+## Motivation
+
+Adding support for package source mapping in PMUI will allow user to make/view package source mappings with more ease. It will also help the user be more aware of where their packages are coming from and what mappings they have already configured
+
+## Explanation
+
+### Functional Explanation
+
+The user will be able to choose whether or not they want to make a mapping to the source in the package source combobox in the details pane. There will be a new checkbox below the package version dropdown which the user can check if they want to map the package they are installing to the source they've selected above. If the user has `All` sources selected, then the checkbox will be grayed out because it doesn't make sense <!--(reword)--> to let the user make mappings from a package to all configured sources.
+ 
+The existing preview dialog that shows when the user hits `Install` or `Update` will be modified to list the mapping they just made under the version. The user already has the option to disable this preview dialog, so if they choose to disable it, there will be nothing telling them what mappings they made.
+
+<!--add screenshot-->
+
+#### Error Messages
+
+There are a few errors the user could make while trying to install or update a package with a mapping that would make restore fail. I will address those errors with a popup telling the user what did not work. First, there could be issues with the transitive packages. Maybe there are transitive packages that are already mapped to a different source or the source that the user is trying to map a package to does not support some of the transitive packages. In either case, a popup would appear telling the user that the package source mapping could not be made. The popup would have a grid with a column showing the top level package and all of the transitive packages, a column showing if each package already has a mapping (if so what source it is mapped to previously), and whether the source supports the package. <!-- <- are all of these things even doable? -->
```

Comment:
Great description. A quick mockup of how this might look would speak loudly! Or even some markdown code block that gives the idea of the error message and empowering action the user can take.